core/templates/dev/head/pages/email_dashboard/EmailDashboardServicesSpec.js

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the email dashboard page.
+ */
+
+describe('Email Dashboard Services', function() {
```

Comment:
Add tests to simulate, e.g., let's say you have 25 queries, try paging forward two times, page backwards once, add a query, page forward again -- make sure the page numbers are correct and that you end up with 26 queries. Also, when fetching a query make sure that the correct one is replaced.

(You'll need to break this down into individual tests, but the upshot is that there ought to be more tests here.)